### PR TITLE
Migrated three j-integral related regression tests to TM

### PR DESCRIPTION
**Pull request by jiangwen84**
The three tests are 

solid_mechanics/test/tests/domain_integral_thermal/j_integral_2d_ctefunc.i
solid_mechanics/test/tests/domain_integral_thermal/j_integral_2d_inst_ctefunc.i
solid_mechanics/test/tests/domain_integral_thermal/j_integral_2d_mean_ctefunc.i

ref #10230

### CODE REVIEW

**Comment by permcody** [comment]
I'm going to assign this to @bwspenc - My only comment on this PR is that I'd prefer to see less usage of "file_base" in test files. We like the connection where the input file name determines the output file name when looking through tests. Using file_base makes it harder to match up inputs and outputs when you aren't familiar with the tests.

File: modules/tensor_mechanics/test/tests/domain_integral_thermal/tests [hunk 1]
```diff
@@ -15,4 +15,19 @@
...
   [../]
+  [./test_jthermal_ctefunc]
+    type = 'CSVDiff'
+    input = 'j_integral_2d_ctefunc.i'
+    csvdiff = 'j_integral_2d_ctefunc_out.csv'
+  [../]
+  [./test_jthermal_mean_ctefunc]
+    type = 'CSVDiff'
+    input = 'j_integral_2d_mean_ctefunc.i'
+    csvdiff = 'mean_out.csv'
```
Comment (bwspenc) [inline modules/tensor_mechanics/test/tests/domain_integral_thermal/tests]: I agree with @permcody: skip the `file_base` commands in the input files, and have this file be `j_integral_2d_mean_ctefunc_out.csv` (the default).

File: modules/tensor_mechanics/test/tests/domain_integral_thermal/tests [hunk 2]
```diff
@@ -15,4 +15,19 @@
...
+  [../]
+  [./test_jthermal_inst_ctefunc]
+    type = 'CSVDiff'
+    input = 'j_integral_2d_inst_ctefunc.i'
+    csvdiff = 'inst_out.csv'
```
Comment (bwspenc) [inline modules/tensor_mechanics/test/tests/domain_integral_thermal/tests]: Same with this one

**Review by bwspenc — CHANGES_REQUESTED** [review]
This looks good.  Just make the changes to the output file names, and this will be ready to go.

**Comment by jiangwen84** [comment]
file_base has been removed :-)

**Review by bwspenc — APPROVED** [review]
Looks good. Thanks for addressing the file_base issue!

**Pull request merged**